### PR TITLE
[ntuple] RTFString: Length is unsigned

### PR DESCRIPTION
**Pull request by hahnjo**
This solves reading strings of more than 128 characters. It is still wrong for longer strings where the first byte is 255 and the following four bytes hold an integer with the actual length.

### CODE REVIEW

**Comment by hahnjo** [comment]
Still a Draft because I need to check tomorrow with a fresh brain how the rest of the code is supposed to work...

**Comment by jblomer** [comment]
Interesting! How did you discover this one?

**Comment by hahnjo** [comment]
> Interesting! How did you discover this one?

That's our segfault with the converted AGC RNTuple files... Eventually I noticed that the offset was going backwards when adding string sizes which didn't sound right :neutral_face:

**Comment by jblomer** [comment]
Wow! Nice work!

**Comment by hahnjo** [comment]
Should be good for review now, working on my end and I added some comments to prevent future developers from being as confused as me...

File: tree/ntuple/v7/test/ntuple_minifile.cxx [hunk 1]
```diff
@@ -324,6 +324,30 @@ TEST(MiniFile, ProperKeys)
    EXPECT_EQ(offset, size);
 }
 
+TEST(MiniFile, LongString)
+{
+   FileRaii fileGuard("test_ntuple_minifile_long_string.root");
+
+   static constexpr const char *LongString =
+      "This is a very long text with exactly 254 characters, which is the maximum that the RNTupleWriter can currently "
```
Comment (jblomer) [inline tree/ntuple/v7/test/ntuple_minifile.cxx:332]: :)

**Review by jblomer — APPROVED** [review]

**Review by vepadulano — APPROVED** [review]
This is great! Thank you so much @hahnjo !

On a separate note, we should consider avoiding writing the full path to the key title otherwise we may not be able to read RNTuples with long paths

**Pull request merged**